packages/front-end/components/Settings/InviteModal.tsx

```diff
@@ -44,29 +47,41 @@ const InviteModal: FC<{ mutate: () => void; close: () => void }> = ({
   }
 
   const onSubmit = form.handleSubmit(async (value) => {
-    const resp = await apiCall<{
-      emailSent: boolean;
-      inviteUrl: string;
-      status: number;
-      message?: string;
-    }>(`/invite`, {
-      method: "POST",
-      body: JSON.stringify(value),
-    });
+    const inviteArr = value.email.split(",");
 
-    if (resp.emailSent) {
-      mutate();
-      close();
-    } else {
-      setInviteUrl(resp.inviteUrl);
-      setEmailSent(resp.emailSent);
-      mutate();
+    if (canSubscribe && activeAndInvitedUsers + inviteArr.length > freeSeats) {
+      setShowUpgradeModal(true);
+      return;
     }
 
-    track("Team Member Invited", {
-      emailSent,
-      role: value.role,
-    });
+    for (const email of inviteArr) {
+      const resp = await apiCall<{
+        emailSent: boolean;
+        inviteUrl: string;
+        status: number;
+        message?: string;
+      }>(`/invite`, {
+        method: "POST",
+        body: JSON.stringify({
+          email: email,
+          role: value.role,
+        }),
+      });
+
+      if (resp.emailSent) {
```

Comment:
We need better logic here when dealing with multiple email addresses.  We should not call `mutate()` and `close()` or any of the `setState` methods within the for loop.  Those should be called once at the end.

Right now, if at least one email succeeds, the modal will immediately close and the user won't be notified about any failures.  And if all of them fail, the `setInviteUrl` will only contain the last one to fail, which isn't very useful.

Reply:
@jdorn Just updated this logic to be a bit more intentional and handle all combinations (that I could think of) of successes/failures.